regimes/mx/complements/fuel_complement.go

```diff
@@ -0,0 +1,58 @@
+// Package complements provides GOBL Complements for Mexican invoices
+package complements
+
+import (
+	"cloud.google.com/go/civil"
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/num"
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/gobl/tax"
+)
+
+// FuelComplement provides detailed information about fuel purchases made with
+// electronic wallets. In Mexico, e-wallet suppliers are required to report this
+// information as part of the invoices they issue to their customers.
+type FuelComplement struct {
+	// Customer's e-wallet account number
+	WalletAccountNumber string `json:"wallet-account-number" jsonschema:"title=Wallet Account Number"`
+	// List of fuel purchases made with the customer's e-wallets
+	Lines []*FuelLine `json:"lines" jsonschema:"title=Lines"`
+	// Summary of all the purchases totals, including taxes (calculated)
+	Totals bill.Totals `json:"totals" jsonschema:"title=Totals" jsonschema_extras:"calculated=true"`
+}
+
+// FuelLine represents a single fuel purchase made with an e-wallet issued by
+// the invoice's supplier to the invoice's customer.
+type FuelLine struct {
```

Comment:
I used the "line" word because the whole structure resembles very much to the `bill.Line` struct. We could even extend the `bill.Line` struct here with the specific fields we need in this context, which enable to potentially reuse part of the validation/calculation logic. However, I thought that binding the evolution of the `bill.Line` struct and behaviour to this struct may not be a good idea if those diverge in the future. Also, like I mentioned in other [comments](https://github.com/invopop/gobl/pull/192/files#r1313014650), `bill.Line` has fields that won't be used here, and that could confuse developers.

Reply:
I think we're okay recreating the structures for this use-case. FuelLine makes sense.